build.gradle.kts

```diff
@@ -180,26 +229,75 @@ subprojects {
     apply(plugin = "net.linguica.maven-settings")
 
     val sourcesJar by tasks.registering(Jar::class) {
-        archiveClassifier.set("sources")
-        from(sourceSets.main.get().allSource)
+        classifier = "sources"
+        from(sourceSets["main"].allSource)
     }
 
     val javadocJar by tasks.registering(Jar::class) {
-        archiveClassifier.set("javadoc")
+        classifier = "javadoc"
         from(tasks["javadoc"])
     }
 
+    val outputDirectories by configurations.creating {
+
+    }
+
+    val testFixturesCompile by configurations.creating {
+        extendsFrom(configurations["implementation"])
+    }
+
+    configurations.create("testFixturesRuntime") {
+        extendsFrom(configurations["runtimeOnly"], configurations["testFixturesCompile"])
+    }
+
+    val testFixturesUsageImplementation by configurations.creating {
+        extendsFrom(configurations["testFixturesCompile"], configurations["outputDirectories"])
+    }
+
+    val testFixturesUsageRuntimeOnly by configurations.creating {
+        extendsFrom(configurations["testFixturesRuntime"], configurations["testFixturesUsageImplementation"])
+    }
+
+    configurations["testImplementation"].extendsFrom(testFixturesUsageImplementation)
+    configurations["testRuntimeOnly"].extendsFrom(testFixturesUsageRuntimeOnly)
+
+    sourceSets.create("testFixtures") {
+        java.srcDir("src/testFixtures/java")
+        resources.srcDir("src/testFixtures/resources")
+        compileClasspath = sourceSets["main"].output + configurations["testFixturesCompile"]
+        runtimeClasspath = output + compileClasspath + configurations["testFixturesRuntime"]
+    }
+
+    dependencies {
+        outputDirectories(sourceSets["testFixtures"].output)
+        testFixturesUsageImplementation(project(project.path))
+        testFixturesCompile("org.slf4j:slf4j-api:${project.extra.get("slf4jVersion")}")
+        testFixturesCompile("junit:junit:${project.extra.get("junitVersion")}")
+        testFixturesCompile(gradleApi())
+    }
+
+    val testFixturesJar by tasks.registering(Jar::class) {
+        classifier = "test-fixtures"
+        from(sourceSets["testFixtures"].output)
+    }
+
+    if (project.name == "common") {
+        artifacts {
+            add("archives", tasks["testFixturesJar"])
+            add("testFixturesCompile", testFixturesJar)
+        }
+    }
```

Comment:
Couple of questions / comments:
1. I guess this is simulating the test-fixture plugin from 5.6. Does it install / publish any form of jar ? 
2. Should it use the original fixtures for >=5.6 ? (I'm inclined to say no especially if this publishes an extra jar with a classifier? ) 

Can we a couple of sentences at the top explaining why we are using this block that looks like 5.6 test fixtures?

Reply:
> 1. I guess this is simulating the test-fixture plugin from 5.6. Does it install / publish any form of jar ?

Yes, common test-fixtures is published before, so I was trying to also publish this one.

> 2. Should it use the original fixtures for >=5.6 ? (I'm inclined to say no especially if this publishes an extra jar with a classifier? )

I don't think we can, in that it's not something we can just `if` out on older versions.

> Can we a couple of sentences at the top explaining why we are using this block that looks like 5.6 test fixtures?

Will do.